tools/Makefile.shared

```diff
@@ -12,10 +12,53 @@
 #*   special exception on linking described in the file LICENSE.          *
 #*                                                                        *
 #**************************************************************************
-
+MAKEFLAGS := -r -R
 include ../config/Makefile
-CAMLRUN ?= ../boot/ocamlrun
-CAMLYACC ?= ../boot/ocamlyacc
+ifndef CAMLRUN
```

Comment:
`$$$$` looks a bit scary. Could you maybe add a comment on what the subtle code below is doing and why it is necessary?